sqlglot/dialects/bigquery.py

```diff
@@ -224,6 +224,24 @@ class Parser(parser.Parser):
             "OPTIONS": lambda self: exp.Properties(expressions=self._parse_with_property()),
         }
 
+        def _parse_table_part(self, schema: bool = False) -> t.Optional[exp.Expression]:
+            this = (
```

Comment:
can you make this a first class function and call it in both places so it's not copy and pasted?

Reply:
sure

Reply:
done